controllers/profiles/getOtherProfileByUsername.js

```diff
@@ -17,20 +17,22 @@ module.exports = async (req, res) => {
       });
     }
 
+    const targetUserId = user?.dataValues?.user_id;
+
     const getFollowerCountQuery = `SELECT COUNT(user_follow_user.user_id_follower) as count_follower from user_follow_user WHERE user_id_followed = :user_id`;
     const getFollowingCountQuery = `SELECT COUNT(user_follow_user.user_id_followed) as count_following from user_follow_user WHERE user_id_follower = :user_id`;
     const isFollowingQuery = `SELECT * FROM user_follow_user WHERE user_id_followed= :user_id_followed AND user_id_follower= :user_id_follower`;
 
     const getFollowerCount = await sequelize.query(getFollowerCountQuery, {
-      replacements: {user_id: req.userId}
+      replacements: {user_id: targetUserId}
     });
 
     const getFollowingCount = await sequelize.query(getFollowingCountQuery, {
-      replacements: {user_id: req.userId}
+      replacements: {user_id: targetUserId}
     });
 
     const isFollowing = await sequelize.query(isFollowingQuery, {
-      replacements: {user_id_follower: req?.userId, user_id_followed: user?.dataValues?.user_id}
+      replacements: {user_id_follower: req?.userId, user_id_followed: targetUserId}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are logical and improve the accuracy of the follower count, following count, and whether the user is being followed. The use of `targetUserId` instead of `req.userId` in the SQL queries ensures that the correct user's data is being retrieved.
```diff
-      replacements: {user_id: req.userId}
+      replacements: {user_id: targetUserId}

-      replacements: {user_id_follower: req?.userId, user_id_followed: user?.dataValues?.user_id}
+      replacements: {user_id_follower: req?.userId, user_id_followed: targetUserId}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->